tests/model_fields/tests.py

```diff
@@ -661,6 +689,15 @@ def test_backend_range_validation(self):
             }
             with self.assertRaisesMessage(ValidationError, expected_message):
                 instance.full_clean()
+            try:
+                with transaction.atomic():
+                    instance.save()
+            except (DataError, IntegrityError):
+                pass  # backends such as Postgres don't fail silently
+            else:
+                qs = self.model.objects.filter(value__lte=min_value)
```

Comment:
No, as far as I know MySQL is the only one that will silently fail. So we could add a flag on the base `DatabaseFeatures` and one more MySQL.

I want to also see if this applies to floating points as well, this will determine the name to either be `truncates_numbers` or `truncates_integers`